src/components/ModalFlows/StakeScreens/StakeScreens.js

```diff
@@ -27,13 +46,27 @@ function StakeScreens({ mode, stakeManagement, stakeActions }) {
             temporatyTrx.current = temporatyTrx.current.concat(intent)
           })
         }
+
         await stakeActions.stake({ amount }, intent => {
-          const trxList = temporatyTrx.current.concat(intent)
-          setTransactions(trxList)
-          onComplete()
+          temporatyTrx.current = temporatyTrx.current.concat(intent)
+
+          if (stakingAllowance?.gt(0)) {
+            setTransactions(temporatyTrx.current)
+            onComplete()
+          }
         })
+
+        if (stakingAllowance?.eq(0)) {
```

Comment:
giving a second thought to this and looking at it i think that is a bit redundat to have this other if here

Reply:
Maybe refactoring it this way might be more clear

```
   await stakeActions.stake({ amount }, intent => {
        temporatyTrx.current = temporatyTrx.current.concat(intent)
      })
    
      if (stakingAllowance?.eq(0)) {
        await stakeActions.allowManager(intent => {
          temporatyTrx.current = temporatyTrx.current.concat(intent)
        })
      }
    
      setTransactions(temporatyTrx.current)
      onComplete()
    
      return
```

Reply:
make sense!